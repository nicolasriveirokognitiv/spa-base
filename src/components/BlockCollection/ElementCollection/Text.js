import React from 'react';
import { toolPropTypes } from 'Constants/propTypes';
import { Input } from 'antd';
import useTestMode from 'Hooks/useTestMode';
import { elementCategories } from 'Constants/global';

function Text({ label, onChange, value, defaultValue, testId }) {
    const testProps = useTestMode(testId);
    const handleChange = e => onChange(e.target.value);

    return (
        <Input
            value={value}
            defaultValue={defaultValue}
            placeholder={label}
            onChange={handleChange}
            {...testProps}
        />
    );
}

Text.propTypes = toolPropTypes;
Text.defaultProps = {
    label       : 'Text',
    propertyKey : 'text',
    category    : elementCategories.CONTENT,
    onChange    : () => {},
};

export default Text;
