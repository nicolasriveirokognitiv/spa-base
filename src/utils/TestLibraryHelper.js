/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import initStore from 'State/store';

const store = initStore();

const renderWithRedux = children => render(
    <Provider store={store}>
        {children}
    </Provider>,
);

export * from '@testing-library/react';
export { renderWithRedux, render };
