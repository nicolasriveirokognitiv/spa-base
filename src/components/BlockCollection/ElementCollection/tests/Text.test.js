import React from 'react';
import { render } from 'Utils/TestLibraryHelper';
import EditorContext from 'State/editorContext';
import Text from '../Text';

describe('Text', () => {
    let container;

    beforeEach(() => {
        container = document.createElement('div');
        document.body.appendChild(container);
    });

    const renderText = props => render(
        <EditorContext.Provider value={{ isTestModeActive: true }}>
            <Text {...props} />
        </EditorContext.Provider>,
    );

    test('Should render Correctly', () => {
        expect(<Text />).toBeDefined();
    });

    test('Should match snapshot', async () => {
        const { asFragment } = render(<Text />);
        expect(asFragment()).toMatchSnapshot();
    });

    test('Text should be recibed on props', () => {
        const { getByTestId } = renderText({ value: 'Text', testId: 'text-input' });
        const elem = getByTestId('text-input');
        expect(elem.value).toBe('Text');
    });
});
