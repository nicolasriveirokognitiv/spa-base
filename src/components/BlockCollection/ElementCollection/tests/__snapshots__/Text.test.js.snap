// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Should match snapshot 1`] = `
<DocumentFragment>
  <input
    class="ant-input"
    placeholder="Text"
    type="text"
    value=""
  />
</DocumentFragment>
`;
